pkg/operator/admission/webhook.go

```diff
@@ -0,0 +1,212 @@
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"os"
+)
+
+const (
+	appName            = "rook-ceph-admission-controller"
+	secretVolumeName   = "webhook-certificates"
+	serviceAccountName = "rook-ceph-admission-controller"
+	storageBackendCeph = "ceph"
+	portName           = "webhook-api"
+)
+
+var (
+	validatePath       = "/validate"
+	servicePort  int32 = 443
+	serverPort   int32 = 8079
+	namespace          = os.Getenv(k8sutil.PodNamespaceEnvVar)
+)
+
+func isSecretPresent(context *clusterd.Context) (bool, error) {
+	logger.Infof("looking for secret %q", appName)
+	_, err := context.Clientset.CoreV1().Secrets(namespace).Get(appName, metav1.GetOptions{})
+	if err != nil {
+		// If secret is not found. All good ! Proceed with rook without admission controllers
+		if apierrors.IsNotFound(err) {
+			logger.Infof("secret %q not found. proceeding without admission controllers", appName)
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func createWebhookService(context *clusterd.Context) error {
+	logger.Infof("creating service %q", appName)
+	logger.Infof("creating in namespace %s", namespace)
+	webhookService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: servicePort,
+					TargetPort: intstr.IntOrString{
+						IntVal: serverPort,
+					},
+				},
+			},
+
+			Selector: map[string]string{
+				k8sutil.AppAttr: appName,
+			},
+		},
+	}
+
+	_, err := k8sutil.CreateOrUpdateService(context.Clientset, namespace, &webhookService)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// InitWebhookIfSecretPresent will initialize the webhook if secret is detected
+func InitWebhookIfSecretPresent(context *clusterd.Context, admissionImage string) error {
+	isPresent, err := isSecretPresent(context)
+	if err != nil {
+		return errors.Wrap(err, "failed to retrieve secret")
+	}
+	if isPresent {
+		err = initWebhook(context, admissionImage)
+		if err != nil {
+			return errors.Wrap(err, "failed to initialize webhook")
+		}
+	}
+	return nil
+}
+
+func initWebhook(context *clusterd.Context, admissionImage string) error {
+	// At this point volume should be mounted, so proceed with creating the service and validatingwebhookconfig
+	err := createWebhookService(context)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service")
+	}
+	err = createWebhookPods(context, admissionImage)
+	if err != nil {
+		return errors.Wrap(err, "failed to create service")
+	}
+	return nil
+}
+
+func createWebhookPods(context *clusterd.Context, admissionImage string) error {
+	logger.Info("creating admission controller pods")
+	logger.Infof("creating in namespace %s", namespace)
```

Comment:
let's remove all the logging for the namespace. If it's still needed, you could modify the line before to include it in the same message.